app/middlewares/vendor_api_request_middleware.rb

```diff
@@ -0,0 +1,57 @@
+require 'redis'
+require './app/workers/vendor_api_request_worker'
+
+class VendorAPIRequestMiddleware
+  REQUEST_HEADER_KEYS = %w[
+    HTTP_VERSION
+    HTTP_HOST
+    HTTP_USER_AGENT
+    HTTP_ACCEPT
+    HTTP_ACCEPT_ENCODING
+    HTTP_AUTHORIZATION
+    HTTP_CONNECTION
+    HTTP_CACHE_CONTROL
+  ].freeze
+
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    started_at = Time.zone.now
+
+    @request = Rack::Request.new(env)
+    status, headers, response = @app.call(env)
+
+    begin
+      if vendor_api_path?
+        finished_at = Time.zone.now
+        response_time = (finished_at - started_at) * 1000
+
+        VendorAPIRequestWorker.perform_async(request_data, response.body, status, response_time, Time.zone.now)
```

Comment:
Can we add a feature flag here so we can quickly disable if this causes trouble?